shoop/campaigns/admin_module/form_sets.py

```diff
@@ -58,11 +58,24 @@ def get_queryset(self):
         return self.owner.conditions.instance_of(self._get_actual_model())
 
 
-class BasketEffectsFormSet(BaseFormset):
+class BasketDiscountEffectsFormSet(BaseFormset):
     model = BasketDiscountEffect
 
     def get_queryset(self):
-        return self.owner.effects.instance_of(self._get_actual_model())
+        return self.owner.basket_discount_effects.instance_of(self._get_actual_model())
+
+    def form(self, **kwargs):
+        if issubclass(self.form_class, TranslatableModelForm):
+            kwargs.setdefault("languages", settings.LANGUAGES)
+            kwargs.setdefault("default_language", settings.PARLER_DEFAULT_LANGUAGE_CODE)
+        return self.form_class(initial={"campaign": self.owner}, **kwargs)
```

Comment:
The language stuff is replication of the super class functionality, just do a proper super call and it won't be needed:
```python
def form(self, **kwargs):
    kwargs.setdefault("initial", {"campaign": self.owner})
    return super(BasketDiscountEffectsFormSet, self).form_class(**kwargs)
```